sdk/cosmosdb/cosmos/src/ClientContext.ts

```diff
@@ -552,14 +575,16 @@ export class ClientContext {
     options: RequestOptions = {}
   ): Promise<Response<DatabaseAccount>> {
     const endpoint = options.urlConnection || this.cosmosClientOptions.endpoint;
-    const request: RequestContext = {
+    const request: WithRequired<RequestContext, "path" | "resourceId" | "resourceType" | "operationType" | "headers"> = {
       ...this.getContextDerivedPropsForRequestCreation(),
       endpoint,
       method: HTTPMethod.get,
+      resourceId: undefined as any,
```

Comment:
is this property going to be updated later? could this object be typed with a more precise type rather than artificially adding stuff to it?

Reply:
Hey deyaaeldeen, This PR is more for showcasing the kind of changes I am proposing for typesafety. I would request to wait till tomorrow when I would be adding a doc, explaining the changes are rational behind them.